elasticapm/base.py

```diff
@@ -496,10 +481,11 @@ def handle_transport_fail(self, exception=None, **kwargs):
     def _collect_transactions(self):
         self._stop_send_timer()
         transactions = []
-        for transaction in self.instrumentation_store.get_all():
-            for processor in self.processors:
-                transaction = processor(self, transaction)
-            transactions.append(transaction)
+        if self.instrumentation_store:
```

Comment:
this `if` is related to one of the subtle bugs you mentioned? isn't possible to init `instrumentation_store` to something not `None`?, like eg. in a way that if you can `get_all()` on it you get an empty list...

Reply:
The problem here was if we had a configuration error, we would bail out early in `__init__`, before the `instrumentation_store` attribute was even set. Then when we try to collect transactions, the code explodes. Due to mocking, that code path was never hit in tests.

I don't think it's worth the effort to initialize the `instrumentation_store` to a `DummyStore` or somesuch, only to change it to the real store a few lines down (if the config validates), tbh.

Reply:
i see, makes sense. thanks!